From 0c4a938ed4a2cd0c2d31c5c616594bae0865d7f3 Mon Sep 17 00:00:00 2001
From: Ernesto Ramos <ernesto@ti.com>
Date: Wed, 28 Jul 2010 09:56:01 -0500
Subject: [PATCH 088/116] staging:ti dspbridge: remove find_gcf from nldr.c

Remove find_gcf from nldr.c and use kernel function
gcd().

Signed-off-by: Ernesto Ramos <ernesto@ti.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---
 drivers/staging/tidspbridge/rmgr/nldr.c |   20 ++------------------
 1 files changed, 2 insertions(+), 18 deletions(-)

diff --git a/drivers/staging/tidspbridge/rmgr/nldr.c b/drivers/staging/tidspbridge/rmgr/nldr.c
index 55dfb98..d8f4eeb 100644
--- a/drivers/staging/tidspbridge/rmgr/nldr.c
+++ b/drivers/staging/tidspbridge/rmgr/nldr.c
@@ -35,6 +35,7 @@
 #include <dspbridge/uuidutil.h>
 
 #include <dspbridge/nldr.h>
+#include <linux/gcd.h>
 
 /* Name of section containing dynamic load mem */
 #define DYNMEMSECT  ".dspbridge_mem"
@@ -304,7 +305,6 @@ static void unload_ovly(struct nldr_nodeobject *nldr_node_obj,
 static bool find_in_persistent_lib_array(struct nldr_nodeobject *nldr_node_obj,
 					 struct dbll_library_obj *lib);
 static u32 find_lcm(u32 a, u32 b);
-static u32 find_gcf(u32 a, u32 b);
 
 /*
  *  ======== nldr_allocate ========
@@ -1887,27 +1887,11 @@ static u32 find_lcm(u32 a, u32 b)
 {
 	u32 ret;
 
-	ret = a * b / find_gcf(a, b);
+	ret = a * b / gcd(a, b);
 
 	return ret;
 }
 
-/*
- * ================ Find GCF (Greatest Common Factor ) ===
- */
-static u32 find_gcf(u32 a, u32 b)
-{
-	u32 c;
-
-	/* Get the GCF (Greatest common factor between the numbers,
-	 * using Euclidian Algo */
-	while ((c = (a % b))) {
-		a = b;
-		b = c;
-	}
-	return b;
-}
-
 #ifdef CONFIG_TIDSPBRIDGE_BACKTRACE
 /**
  * nldr_find_addr() - Find the closest symbol to the given address based on
-- 
1.7.1

